pkg/server/api/api.go

```diff
@@ -24,3 +28,23 @@ func ProtoFromID(id spiffeid.ID) *types.SPIFFEID {
 		Path:        id.Path(),
 	}
 }
+
+type AuthorizedEntryFetcher interface {
+	FetchAuthorizedEntries(ctx context.Context) ([]*types.Entry, error)
+}
+
+// FetchAuthEntries fetches authorized entries using caller ID from context
+func FetchAuthEntries(ctx context.Context, log logrus.FieldLogger, ef AuthorizedEntryFetcher) (map[string]*types.Entry, error) {
+	entries, err := ef.FetchAuthorizedEntries(ctx)
```

Comment:
we need to put callerID validation in some place, is the idea to move it into `ef.FetchAuthorizedEntries(ctx)`? if that is the case can you add comments to that interface so, when we we implement it we didnt miss that validation?
annd I think we must validate how our code works when that validation fails.

Reply:
The current implementation gets the caller ID in the `fetchEntries` function and sends the caller ID as an argument to the `FetchAuthorizedEntries` function along with the context, which seems to be redundant, since the `FetchAuthorizedEntries` function can get the caller ID from the context and should return the authorized entries based on that caller ID (current code calls `rpccontext.CallerID(ctx)` twice).
I was thinking that it should be documented that `FetchAuthorizedEntries` should do the proper caller ID validation. I'll add comments to have this documented.

Reply:
In my opinion, use of `rpccontext` should stay at the API layer. I'd rather the dependencies receive their arguments explicitly.